### PR TITLE
Enable code coverage reporting for Spring

### PR DESCRIPTION
**Pull request by yiliuTo**
Follow up PR #18221 to generate code coverage report for Spring.
All modules of starters, samples and tests are excluded for code coverage.

### CODE REVIEW

File: sdk/spring/pom.xml [hunk 1]
```diff
@@ -87,4 +87,141 @@
     <module>azure-spring-cloud-stream-binder-test</module>
```
Comment (srnagar) [inline sdk/spring/pom.xml:87]: All these modules have to be moved to the default profiles section. There should be no modules outside of the profiles section.

**Comment by yiliuTo** [comment]
Hi @srnagar @saragluna @chenrujun , please help to review this PR. 
Thanks

File: sdk/spring/pom.xml [hunk 2]
```diff
@@ -87,4 +87,141 @@
...
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>default</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <modules>
+        <module>azure-spring-boot</module>
```
Comment (srnagar) [inline sdk/spring/pom.xml:210]: All the 75 modules that are outside should be moved to this `modules` sections

Comment (yiliuTo) [reply sdk/spring/pom.xml:210]: Oh, I didn't see the default profile. Will update it soon.

**Review by srnagar — APPROVED** [review]
lgtm!

**Comment by yiliuTo** [comment]
@chenrujun @saragluna , should `azure-spring-cloud-messaging` be included into code coverage report? I find that there is no jacoco coverage limit in its pom.xml.

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
https://dev.azure.com/azure-sdk/public/_build/results?buildId=663869&view=codecoverage-tab

**Comment by saragluna** [comment]
@yiliuTo I think it should be included. But what do you mean by saying `no jacoco coverage limit`?

**Comment by saragluna** [comment]
@yiliuTo I'll merge this PR to unblock the spring ci pipeline.

**Comment by chenrujun** [comment]
@saragluna , I think she means contents like:
```
  <properties>
    <jacoco.min.linecoverage>0.09</jacoco.min.linecoverage>
    <jacoco.min.branchcoverage>0.14</jacoco.min.branchcoverage>
  </properties>
```

**Comment by saragluna** [comment]
@yiliuTo if this is what you meant, these properties were added as an exception in cases when the specific module's code coverage doesn't meet the minimum requirement.

**Pull request merged**